### ISSUE TITLE
Create a Rocket Details Page

### ISSUE THREAD

**Issue by pushpinderpalsingh** [pushpinderpalsingh/SpaceDash#23]
I am attaching some UI Mockups, which should be opened when someone taps on the rocket on the home page. There should be a little animation that will extend the white part and everything should dissolve in.

![iPhone XS](https://user-images.githubusercontent.com/47032662/94310858-ee37cb80-ff97-11ea-8185-9eb8ac4d8f42.png) ![iPhone XS Read More](https://user-images.githubusercontent.com/47032662/94310843-ebd57180-ff97-11ea-8b05-93934a4a795f.png)

The second screen should open when someone taps on the read more button. 

The first order of business is to create the UI, animation can be configured later. Use dummy data for now, we will integrate it with API afterward.

PS: I am not the best designer, if you have some ideas to better the UI or animation, let us discuss in the comments.

**Comment by devwaseem** [comment]
Hi. I'll work on this

**Comment by pushpinderpalsingh** [comment]
Sure thing. I am assigning it to you.

**Comment by pushpinderpalsingh** [comment]
@devwaseem Can you confirm if you are still working on this issue?

**Comment by dheerajghub** [comment]
Can we use programmatic UI to achieve this?

**Comment by pushpinderpalsingh** [comment]
I generally prefer but you definitely can use programmatic UI. Just make sure you have properly documented the code. I would recommend [this video]( https://www.youtube.com/watch?v=hu90G_M0vaA) for documenting your code.

**Comment by dheerajghub** [comment]
Yeah sure! I'll do this. Can you assign it to me? And can I add my own design ideas in it?

**Comment by pushpinderpalsingh** [comment]
if the design ideas are minor it will be fine. But if there is anything major, provide mockups of the changes in the comments of this issue before implementing code. @dheerajghub

**Comment by pushpinderpalsingh** [comment]
Sure thing. I am assigning this to you. While you are here, do 🌟  the repo!

**Comment by dheerajghub** [comment]
Look at these designs do you find these relevant, I've already done with the first part 😅  but the second one is just mock up! Any modifications you want me to do! Once you acknowledge me that this is relevant and good I'll proceed further.

![](https://imgur.com/fNeexeK.png) ![](https://imgur.com/ItvHwR7.png)

**Comment by pushpinderpalsingh** [comment]
@dheerajghub I don't think that would work. I have some metrics that this design is not. Also, I have plans to integrate AR on tap action of Rockets. 

On second thoughts, if you can somehow integrate all the metrics in the little pop-up, while not overcrowding it and having an option to add a button for AR in the future that would be great.

**Comment by dheerajghub** [comment]
AR Button can be placed on the rocket detail page, If you want that! BTW if we implement two popups in one view controller that sure will become messy!

**Comment by pushpinderpalsingh** [comment]
I am not saying two popups. I am saying add more metrics to the one you have made for Description. After a line or two of description, add some metrics like present in my design. Which would increase the pop-up's height.

**Comment by dheerajghub** [comment]
With metrics you mean data? And what about AR Button?

**Comment by pushpinderpalsingh** [comment]
Let's discuss this over slack!

**Comment by dheerajghub** [comment]
This is my final work what do you think?

<img src="https://imgur.com/eRukr7T.png" height = "400">

**Comment by pushpinderpalsingh** [comment]
Personally, I don't like it. The initial mockup was according to me more inclusive of the current design. Can you let me know whats wrong with my design?

**Comment by dheerajghub** [comment]
In your second mockup, there is a popUp view which contains a description of the rocket what if the description is too long this is not a great idea to show it as a popup! What's wrong about my design?? Give feedback to both views. Please!

**Comment by pushpinderpalsingh** [comment]
It's ok to create a pop-up as you designed for the read more. If the description is too long, it should adjust accordingly. It's not a good option to leave a full view empty "if" description is long. 

As for your implementation, I don't like the design. Plus, if you have actually looked at my design it's not a pop-up, its a separate view. In your design, we would be having the pop-up inside of pop-ups I don't think that's a good idea.

**Comment by dheerajghub** [comment]
#62

**Comment by pushpinderpalsingh** [comment]
@dheerajghub you will need to open a new pr as pr #62  was not building successfully and is reverted back.

**Comment by pushpinderpalsingh** [comment]
@dheerajghub I am going to unassign this issue because of inactivity.

**Comment by anubhavsingh16** [comment]
@pushpinderpalsingh im working on this issue can you please assign to me?

**Comment by pushpinderpalsingh** [comment]
Sure thing. I am assigning this to you.
